src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -328,6 +328,13 @@ private void validateClusteringOrder(@Nullable Schema schema, FailureCollector c
     if (!shouldCreatePartitionedTable() || Strings.isNullOrEmpty(clusteringOrder) || schema == null) {
       return;
     }
+
+    if(!Strings.isNullOrEmpty(clusteringOrder) && Strings.isNullOrEmpty(partitionByField)){
+      collector.addFailure(String.format("Clustering order cannot be validated."),
```

Comment:
What happens if clusteringOrder is specified, and partitionByField is a macro? Can you test that and make sure it works as expected?

Reply:
Fixed